darwin/config.py

```diff
@@ -4,61 +4,107 @@
+
 
 class Config(object):
     """Handles YAML configuration files"""
 
-    def __init__(self, path: Union[Path, str], data: Optional[dict] = None):
+    def __init__(self, path: Union[Path, str]):
         if isinstance(path, str):
             path = Path(path)
         self._path = path
+        self._data = self._parse()
 
-        if data is None:
-            self._data = self.parse()
-        else:
-            self._data = data
-            self._save()
+    def _parse(self):
+        """Parses the YAML configuration file"""
+        try:
+            with open(self._path, "r") as stream:
+                return yaml.safe_load(stream)
+        except FileNotFoundError:
+            return {}
 
-    def get(self, key: Union[str, List[str]], acc: dict = None) -> Any:
+    def get(self, key: Union[str, List[str]], default: Optional[any] = None) -> Any:
         """Gets value defined by key
         
         Args:
         - key: the key where the value to be fetched is stored.
         It can be formatted as a simple string, or as a path/like/string to fetch nested values.
         """
 
-        if acc is None:
-            acc = self._data.copy()
+        acc = self._data.copy()
 
-        if isinstance(key, str):
-            key = key.split("/")
-        key, *keys = key
-        if len(keys) == 0:
-            return acc.get(key)
-        return self.get(keys, acc.get(key))
+        while True:
+            if isinstance(key, str):
+                key = key.split("/")
+            key, *keys = key
+            acc = acc.get(key)
+            if acc is None:
+                return default
+            if len(keys) == 0:
+                return acc
+            else:
+                key = keys
 
-    def parse(self):
-        """Parses the YAML configuration file"""
-        with open(self._path, "r") as stream:
-            return yaml.safe_load(stream)
-
-    def write(self, key: Union[str, List[str]], value: Any, save: bool = True):
+    def put(self, key: Union[str, List[str]], value: any, save: bool = True):
         """Sets value for specified key
         
         Args:
         - key: the key where the value is going to be stored.
         It can be formatted as a simple string, or as a path/like/string to fetch nested values.
         - value: the value to be set"""
-
         if isinstance(key, str):
             key = key.split("/")
+
+        pointer = self._data
+
         for k in key[:-1]:
-            self._data = self._data.setdefault(k, {})
-        self._data[key[-1]] = value
+            pointer = pointer.setdefault(k, {})
+        pointer[key[-1]] = value
+
         if save:
             self._save()
 
     def _save(self):
         """Persist the configuration to the file system"""
         with io.open(self._path, "w", encoding="utf8") as f:
             yaml.dump(self._data, f, default_flow_style=False, allow_unicode=True)
+
+    def set_team(self, team: str, api_key: str):
```

Comment:
Proposed renaming: `set_team_api_key`

Reply:
it now takes one more argument so I think we can keep it as set_team